core/src/main/scala/org/apache/spark/scheduler/Stage.scala

```diff
@@ -110,3 +133,8 @@ private[spark] abstract class Stage(
     case _ => false
   }
 }
+
+private[scheduler] object Stage {
+  // The number of consecutive failures allowed before a stage is aborted
+  val MAX_CONSECUTIVE_FAILURES = 4
```

Comment:
this is really `MAX_CONSECUTIVE_FETCH_FAILURES`; we don't have use this cap for other kinds of failures.